flink-runtime/src/test/java/org/apache/flink/runtime/rest/handler/legacy/backpressure/BackPressureStatsTrackerImplTest.java

```diff
@@ -18,166 +18,218 @@
 
-		int sampleId = 1231;
-		int endTime = 841;
+		// trigger back pressure stats request
+		tracker.getOperatorBackPressureStats(executionJobVertex);
 
-		StackTraceSample sample = new StackTraceSample(
-				sampleId,
-				0,
-				endTime,
-				traces);
+		Optional<OperatorBackPressureStats> optionalStats = tracker.getOperatorBackPressureStats(executionJobVertex);
+		assertTrue(optionalStats.isPresent());
+		OperatorBackPressureStats stats = optionalStats.get();
 
-		// Succeed the promise
-		sampleFuture.complete(sample);
+		checkOperatorBackPressureStats(taskVertices, requestId, endTime, backPressureRatio, stats);
+	}
 
-		Assert.assertTrue(tracker.getOperatorBackPressureStats(jobVertex).isPresent());
+	@Test
+	public void testOperatorBackPressureStatsUpdate() throws Exception {
+		final ExecutionJobVertex jobVertex = BackPressureTrackerTestUtils.createExecutionJobVertex();
+		final int backPressureStatsRefreshInterval = 2000;
+		final long waitingTime = backPressureStatsRefreshInterval + 500;
+
+		final int requestId1 = 0;
+		final long startTime1 = System.currentTimeMillis();
+		final long endTime1 = startTime1 + 1;
+		final double backPressureRatio1 = 0.1;
+
+		final int requestId2 = 1;
+		final long startTime2 = System.currentTimeMillis() + waitingTime;
+		final long endTime2 = startTime2 + 1;
+		final double backPressureRatio2 = 0.2;
+
+		final BackPressureStats backPressureStats1 = createBackPressureStats(
+			jobVertex.getTaskVertices(), requestId1, startTime1, endTime1, backPressureRatio1);
+		final BackPressureStats backPressureStats2 = createBackPressureStats(
+			jobVertex.getTaskVertices(), requestId2, startTime2, endTime2, backPressureRatio2);
+		final BackPressureStatsTracker tracker = createBackPressureTracker(
+			600000, backPressureStatsRefreshInterval, backPressureStats1, backPressureStats2);
+
+		// trigger back pressure stats request
+		assertFalse(tracker.getOperatorBackPressureStats(jobVertex).isPresent());
+
+		Optional<OperatorBackPressureStats> optionalStats = tracker.getOperatorBackPressureStats(jobVertex);
+		assertTrue(optionalStats.isPresent());
+		OperatorBackPressureStats stats = optionalStats.get();
+
+		final int requestId = stats.getRequestId();
+		checkOperatorBackPressureStats(jobVertex.getTaskVertices(), requestId1, endTime1, backPressureRatio1, stats);
+
+		// should not trigger new back pressure stats request
+		assertTrue(tracker.getOperatorBackPressureStats(jobVertex).isPresent());
+		assertEquals(requestId, tracker.getOperatorBackPressureStats(jobVertex).get().getRequestId());
+
+		// ensure that we are ready for next request
+		Thread.sleep(waitingTime);
+
+		// trigger next back pressure stats request
+		assertTrue(tracker.getOperatorBackPressureStats(jobVertex).isPresent());
+		OperatorBackPressureStats newStats = tracker.getOperatorBackPressureStats(jobVertex).get();
+		assertNotEquals(requestId, newStats.getRequestId());
```

Comment:
Does it exist some unstable issue here that the previous future has not completed by another thread when we get `newStats` here?
It seems better to use `assertEquals(requestId2, newStats.getRequestId())` instead.